[added] Live Schedule filters by event type

dtsivkovski opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have tested my changes locally and they work
- [x] My pull request targets the `main` branch of this repository.
- [x] I have organized any project files to the best of my ability

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #105

<!-- Feel free to add any additional description of changes below this line -->

This pull request introduces a new feature to filter the schedule by activity type and includes several UI improvements in the `ScheduleSection` component. The most important changes involve adding a `ScheduleFilters` component, integrating it into the schedule, and enhancing the layout and interactivity of the schedule UI.

### New Feature: Schedule Filters
* Added a new `ScheduleFilters` component (`app/(event)/live/_components/schedule-filters.tsx`) that allows users to toggle filters for different activity types (e.g., "Check-in," "Main Event"). The component dynamically updates the active filters using state management.
* Integrated the `ScheduleFilters` component into the `ScheduleSection` by importing it and managing its state (`activeFilters` and `setActiveFilters`) in the parent component. [[1]](diffhunk://#diff-f529b55f417aec5c23b8598505124966b3f8c147ed05697cb3edb263aa0e157fR6) [[2]](diffhunk://#diff-f529b55f417aec5c23b8598505124966b3f8c147ed05697cb3edb263aa0e157fR16)

### UI Improvements
* Updated the layout of the `ScheduleSection` to center its content and improve spacing, making the UI more visually appealing.
* Enhanced the "Refresh Schedule" button with a ghost variant and hover effects for a more polished user experience.
* Modified the event rendering logic to respect the active filters, ensuring only events matching the selected filters are displayed. [[1]](diffhunk://#diff-f529b55f417aec5c23b8598505124966b3f8c147ed05697cb3edb263aa0e157fL117-R135) [[2]](diffhunk://#diff-f529b55f417aec5c23b8598505124966b3f8c147ed05697cb3edb263aa0e157fL136-R146)

### Screenshots
![image](https://github.com/user-attachments/assets/32dd9cbd-2049-4bb3-9a23-a2584988ec44)
![image](https://github.com/user-attachments/assets/e56b944d-d969-4588-a767-6af43a958b61)
![image](https://github.com/user-attachments/assets/a935ff14-1118-4a98-b1ab-4015322fc48b)

DylanDevelops reviewed (changes requested):
This is a really good start but before we merge this, we have to fix some issues.

1) Since you are setting the `key` property of `<EventCard />` to the activity type, multiple instances of `<EventCard />` may have the same key value. These key values need to be unique and thus it throws an error such as this:
```
Encountered two children with the same key, `Pizza Party`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
```

2) By default, make all the filters checked correctly showing that all the filters are toggled on.

3) When changing a filter, it should update automatically and not require a refresh as it makes it confusing to use.

3) The refresh button, if we go with the icon design like that, I think should sit on the left side of the "Last refreshed" text. Additionally, I still think clickable buttons like that should be the red button we have for the add to calendar button as well as all over the website. You could also potentially put the last refreshed text inside of the button so that the icon is on the left and the text is on the right and the whole thing is clickable. Let me know if you agree.

Once these have all been implemented, we should be good to go.

dtsivkovski commented:
> This is a really good start but before we merge this, we have to fix some issues.
> 
> 1. Since you are setting the `key` property of `<EventCard />` to the activity type, multiple instances of `<EventCard />` may have the same key value. These key values need to be unique and thus it throws an error such as this:
> 
> ```
> Encountered two children with the same key, `Pizza Party`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
> ```
> 
> 2. By default, make all the filters checked correctly showing that all the filters are toggled on.
> 3. When changing a filter, it should update automatically and not require a refresh as it makes it confusing to use.
> 4. The refresh button, if we go with the icon design like that, I think should sit on the left side of the "Last refreshed" text. Additionally, I still think clickable buttons like that should be the red button we have for the add to calendar button as well as all over the website. You could also potentially put the last refreshed text inside of the button so that the icon is on the left and the text is on the right and the whole thing is clickable. Let me know if you agree.
> 
> Once these have all been implemented, we should be good to go.

Thanks for your feedback, Dylan! I have looked back and implemented some changes.

1. I have fixed the key property and have changed the key back to utilizing an index. Copilot originally suggested changing it to be a key by the name of your event, but you're right that we could have duplicate names that I didn't consider, so thank you for the suggestion.

2. I appreciate this suggestion, but I have already consulted with the designers on this part and have gathered feedback, and the general consensus is that having all of them toggled on by default makes the most sense to the user.

3. I have not experienced this issue. I would love to fix this, but I can't seem to recreate it. Could you explain more in detail about what exactly is happening and what steps I can take to recreate it? The filters apply automatically for me.

4. Personally, I disagree. I think the option to refresh does not visually need to call out the user to click it with a bright red button, and should just be a subtle thing. If we want to make it a ghost button or an outlined button, I would not be opposed to it, but I think highlighting in red unnecessarily draws the user's attention to something that isn't necessarily important. I'd recommend we consult the design team about this so that we can get something solidified, as I don't want to be in disagreement about this.

DylanDevelops reviewed (approved):
LGTM! 👍🏼🎉
